plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php

```diff
@@ -132,6 +153,52 @@ public function register_routes() {
 		);
 	}
 
+	public function get_shipping_methods( $request ) {
+		$order = $this->get_object( (int) $request['id'] );
+
+		if ( ! $order || 0 === $order->get_id() ) {
+			return new WP_Error( "woocommerce_rest_{$this->post_type}_invalid_id", __( 'Invalid ID.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		$cart = new WC_Cart();
+		foreach ( $order->get_items( 'line_item' ) as $item ) {
```

Comment:
We should empty the cart before adding order items to prevent an already non-empty cart interfering with shipping calculations

Reply:
The cart only exists in memory, so for the API it will be destroyed after each request. Granted, unexpected things could happen if we use this code in WP-CLI, but that seems unlikely for a REST API endpoint?

https://github.com/woocommerce/woocommerce/pull/30855/files#diff-0546b54e79a59d3598b80bb065d59cd9dcc3ecb96f7618aba35c33805043cfa0R34